src/common/client_response_texts.cpp

```diff
@@ -3,45 +3,51 @@
 
 #include "client_response_texts.hpp"
 #include "i18n.h"
-//todo make some automatic checks names vs enum
-//list of all button types
-// order and count must match to enum class Response!
-const std::array<const char *, static_cast<size_t>(Response::_last) + 1> BtnTexts::texts {
-    "",                   // _none
-    N_("ABORT"),          // Abort
-    "ABS",                // ABS filament, do not translate
-    "ASA",                // ASA filament, do not translate
-    N_("BACK"),           // Back
-    N_("CANCEL"),         // Cancel
-    N_("CHANGE"),         // Change
-    N_("CONTINUE"),       // Continue
-    N_("COOLDOWN"),       // Cooldown
-    N_("DISABLE SENSOR"), // Filament_removed
-    "FLEX",               //FLEX filament, do not translate
-    "HIPS",               //HIPS filament, do not translate
-    N_("IGNORE"),         // Ignore
-    N_("LOAD"),           // Load
-    N_("NEXT"),           // Next
-    N_("NO"),             // No
-    N_("OK"),             // Ok
-    "PC",                 // PC filament, do not translate
-    "PETG",               // PETG filament, do not translate
-    "PLA",                // PLA filament, do not translate
-    "PP",                 // PP filament, do not translate
-    N_("PURGE MORE"),     // Purge_more
-    N_("REHEAT"),         // Reheat
-    N_("RETRY"),          // Retry
-    N_("STOP"),           // Stop
-    N_("UNLOAD"),         // Unload
-    N_("YES"),            // Yes
+#include "resource.h"
+
+// clang-format off
+/**
+ * @brief list of all button types
+ * order and count must match to enum class Response in general_response.hpp!
+ * TODO make some automatic checks names vs enum
+ */
+const std::array<std::pair<const char *, uint16_t>, static_cast<size_t>(Response::_last) + 1> BtnResponse::texts_and_icons {
+    std::make_pair( "", IDR_PNG_question_48px ),                  // _none
+    std::make_pair( N_("ABORT"), IDR_PNG_disconnect_48px ),       // Abort
```

Comment:
I'd like to see some verification, that the construction of this whole array gets done at compile time.
`std::make_pair` has a constexpr variant in C++14, so theoretically the compiler should be able to do it.

Reply:
I could move it to header to and make it constexpr, but it would consume more codesize (tested)
I have disassembled the code it seems fine to me as it is

Reply:
ok